### PR TITLE
user toggle for menuhiding on mobile and desktop

### PR DESCRIPTION
**Pull request by dillchen**
### Link to Issue
Closes: #4115

### Description of Changes
This PR adds the functionality to respect the user's preference for the sidebar visibility across different screen sizes. If the user has not toggled the sidebar visibility, the app will continue to show or hide the sidebar based on the window size.

Here is a video of the functionality https://www.loom.com/share/273d33aa385143c8b308fa43d27bdef3?from_recorder=1&focus_title=1

### Test Plan
1. Manually tested the sidebar visibility toggle on different screen sizes in both desktop and mobile views.

### Deployment Plan
1. Merge the PR.
2. Deploy the changes to staging for further testing.
3. If everything works as expected, deploy the changes to production.

### Other Considerations
- Check if product wants this (I think we do), it aids readability
- Separate ticket for the community icon flashing (it's a bug, that exists on prod)
- Product / Design Discussion on which menus should we hide? (right now the behavior is both), 
--> we should either hide both
--> we should hide only the in community sidebar (and then we should auto hide the "Quick Switcher" while on Mobile) (I prefer the latter)
- Product / Design Discussion => as we think about "single player mode", our behavior is not to have a "In Community Sidebar" while outside of a community => I'm not sure this makes sense anymore (especially if we allow a toggle on both, more discussion is necessary on this) => Some things that I'd want to be able to access "outside of a community" -> My Drafts, Different ways to toggle the feed, and we'd likely better be able to manage / triage notifications (instead of having them default hide in the top navigation menu in a popover

### CODE REVIEW

**Comment by jnaviask** [comment]
Something broke unit tests?

**Comment by dillchen** [comment]
@meeshlin's design review => We should (ticket separately) that icon stays present

**Comment by dillchen** [comment]
```
Run yarn --cwd packages/commonwealth unit-test
yarn run v1.22.19
$ NODE_ENV=test ts-mocha --project tsconfig.json ./test/unit/**/*.spec.ts
Applied changes: logLevel=Debug to log groups 'all'.
Error: Cannot find module 'mixins/breakpoints.scss'
Require stack:
- /home/runner/work/commonwealth/commonwealth/packages/commonwealth/client/scripts/views/components/component_kit/helpers.ts
- /home/runner/work/commonwealth/commonwealth/packages/commonwealth/client/scripts/state/ui/sidebar/sidebar.ts
- /home/runner/work/commonwealth/commonwealth/packages/commonwealth/client/scripts/state/ui/sidebar/index.ts
- /home/runner/work/commonwealth/commonwealth/packages/commonwealth/test/unit/state/sidebar.spec.ts
- /home/runner/work/commonwealth/commonwealth/node_modules/mocha/lib/mocha.js
- /home/runner/work/commonwealth/commonwealth/node_modules/mocha/lib/cli/one-and-dones.js
```

TLDR Unit Test Breaking because mixin relative import not respected?

----
```
It looks like the test runner is having trouble resolving the mixins/breakpoints.scss import in the helpers.ts file. Since the test runner doesn't need to process this SCSS file, you can mock the import to avoid the error.

Create a new file named mocks.js in your project root or test folder with the following content:
javascript
Copy code

// mocks.js

// Mock SCSS imports for tests
require.extensions['.scss'] = () => {};
Update your unit-test script in your package.json file to include the --require ./mocks.js flag (replace ./mocks.js with the path to your mocks.js file if it's not in the project root). Your unit-test script should now look like this:
json
Copy code

"unit-test": "NODE_ENV=test ts-mocha --project tsconfig.json --require ./mocks.js ./test/unit/**/*.spec.ts"
Now, run the yarn --cwd packages/commonwealth unit-test command again. The test runner should no longer have issues with the SCSS import and should execute the tests successfully.
```

**Review by jnaviask — APPROVED** [review]
Seems good, but
- Need to fix unit test.
- Still need to address behavior where, on mobile, clicking a link from the "sidebar" doesn't close the sidebar (so you need to manually unhide the sidebar after selecting e.g. your topic).

**Comment by zakhap** [comment]
Screen Shot 2023-06-13 at 7 01 57 PM
Screen Shot 2023-06-13 at 7 02 01 PM

I'm not a fan of the mobile experience of the toggle button moving locations.

**Comment by CowMuon** [comment]
Unfortunately, 
1. minor merge conflict 
2. failing tests
3. Product not a fan of approach 
As such bouncing to draft, let's align on solution for #4115 that Product can get behind.
